sdk/storage/azure-storage-common/src/main/java/com/azure/storage/common/policy/RequestRetryPolicy.java

```diff
@@ -108,100 +108,100 @@ stream, the buffers that were emitted will have already been consumed (their pos
+
+                Flux<ByteBuffer> responseBody = response.getBody();
+                if (responseBody == null) {
+                    return attemptAsync(context, next, originalRequest, newConsiderSecondary, newPrimaryTry,
+                        attempt + 1);
                 } else {
-                    action = "NoRetry: Unknown error";
+                    return response.getBody()
+                        .ignoreElements()
+                        .then(attemptAsync(context, next, originalRequest, newConsiderSecondary, newPrimaryTry,
+                            attempt + 1));
                 }
 
-                if (action.charAt(0) == 'R' && attempt < requestRetryOptions.getMaxTries()) {
-                        /*
-                        We increment primaryTry if we are about to try the primary again (which is when we
-                        consider the secondary and tried the secondary this time (tryingPrimary==false) or
-                        we do not consider the secondary at all (considerSecondary==false)). This will
-                        ensure primaryTry is correct when passed to calculate the delay.
-                         */
-                    int newPrimaryTry = (!tryingPrimary || !considerSecondary) ? primaryTry + 1 : primaryTry;
-                    return attemptAsync(context, next, originalRequest, considerSecondary, newPrimaryTry, attempt + 1);
-                }
-                return Mono.error(throwable);
-            });
+            }
+            return Mono.just(response);
+        }).onErrorResume(throwable -> {
+            /*
+             * It is likely that many users will not realize that their Flux must be replayable and get an error upon
+             * retries when the provided data length does not match the length of the exact data. We cannot enforce the
+             * desired Flux behavior, so we provide a hint when this is likely the root cause.
+             */
+            if (throwable instanceof IllegalStateException && attempt > 1) {
+                return Mono.error(new IllegalStateException("The request failed because the size of the contents of "
+                    + "the provided Flux did not match the provided data size upon attempting to retry. This is likely "
+                    + "caused by the Flux not being replayable. To support retries, all Fluxes must produce the same "
+                    + "data for each subscriber. Please ensure this behavior.", throwable));
+            }
+
+            /*
+             * IOException is a catch-all for IO related errors. Technically it includes many types which may not be
+             * network exceptions, but we should not hit those unless there is a bug in our logic. In either case, it is
+             * better to optimistically retry instead of failing too soon. A Timeout Exception is a client-side timeout
+             * coming from Rx.
```

Comment:
'Rx' ?

Reply:
Oh man that's old. A lot of this code was taken from v11, which was built on RxJava, a precursor to Reactor. This doc must've not been updated.